<?xml version="1.0"?>

<!--
  This file is part of the quadrotors_landing_vision distribution (https://github.com/dimianx/quadrotors_landing_vision).
  Copyright (c) 2024-2025 Dmitry Anikin <dmitry.anikin@proton.me>.

  This program is free software: you can redistribute it and/or modify  
  it under the terms of the GNU General Public License as published by  
  the Free Software Foundation, version 3.

  This program is distributed in the hope that it will be useful, but 
  WITHOUT ANY WARRANTY; without even the implied warranty of 
  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU 
  General Public License for more details.

  You should have received a copy of the GNU General Public License 
  along with this program. If not, see <http://www.gnu.org/licenses/>.
-->


<launch>

  <arg name="world_name" default="sim.world"/>
  <arg name="positioning" default="aruco"/>
  <arg name="attempts_count" default="1"/>
  <arg name="output" default="results.csv"/>

  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find quadrotors_landing_vision)/worlds/$(arg world_name)"/>
    <arg name="paused" value="false"/>
    <arg name="use_sim_time" value="true"/>
    <arg name="gui" value="true"/>
    <arg name="headless" value="false"/>
    <arg name="verbose" value="false"/>
  </include>

  <include file="$(find quadrotors_landing_vision)/launch/spawners/spawn_iris.launch">
    <arg name="x" value="0.0" />
    <arg name="y" value="0.0" />
    <arg name="z" value="0.1" />
  </include>

  <include file="$(find quadrotors_landing_vision)/launch/spawners/spawn_husky.launch">
    <arg name="robot_namespace" value="/husky" />
    <arg name="x" value="0" />
    <arg name="y" value="-8" />
    <arg name="z" value="0.1" />
    <arg name="yaw" value="3.14" />
  </include>

  <group if="$(eval positioning == 'aruco')">
    <include file="$(find quadrotors_landing_vision)/launch/positioning/aruco_estimator.launch"/> 
  </group>

  <group if="$(eval positioning == 'uwb')">
    <include file="$(find quadrotors_landing_vision)/launch/sensor_sim/uwb_simulator.launch"/>
    <include file="$(find quadrotors_landing_vision)/launch/positioning/uwb_lqr_vanc_estimator.launch"/> 
  </group>

 <include file="$(find quadrotors_landing_vision)/launch/drone/drone_fsm.launch">
    <arg name="attempts_count" value="$(arg attempts_count)"/>
    <arg name="output" value="$(arg output)"/>
  </include>

  <include file="$(find quadrotors_landing_vision)/launch/ugv/ugv_control.launch">
  </include>

  <group if="$(eval positioning == 'aruco')">
    <include file="$(find quadrotors_landing_vision)/launch/drone/landing_controller.launch">
      <arg name="pose_topic" value="/positioning/aruco"/>
    </include> 
  </group>

  <group if="$(eval positioning == 'uwb')">
    <include file="$(find quadrotors_landing_vision)/launch/drone/landing_controller.launch">
      <arg name="pose_topic" value="/positioning/uwb_lqr_vanc"/>
    </include> 
  </group>

  <node name="rqt_image_view" pkg="rqt_image_view" type="rqt_image_view" output="screen" args="/iris_bottom_fpv/usb_cam/image_info"/>

</launch>
